Tests/L0/Maven/_suite.ts

```diff
@@ -29,6 +29,48 @@ function setupDefaultMavenTaskRunner():tr.TaskRunner {
     return taskRunner;
 }
 
+// Recursively lists all files within the target folder, giving their full paths.
+function listFolderContents(folder):string[] {
+    var result:string[] = [];
+    var filesInFolder = fs.readdirSync(folder);
+
+    filesInFolder.forEach(function (fileInFolder) {
+        result.push(path.join(folder, fileInFolder));
+        if (fs.statSync(path.join(folder, fileInFolder)).isDirectory()) {
+            result = result.concat(listFolderContents(path.join(folder, fileInFolder)));
+        }
+    });
+
+    return result;
+}
+
+// Adds mock exist, checkPath, rmRF and mkdirP responses for given file paths.
+// Takes an object to add to and an array of file paths for which responses should be added.
+// Modifies and returns the argument object.
+function setupMockResponsesForPaths(responseObject:any, paths: string[]) { // Can't use rest arguments here (gulp-mocha complains)
+
+    // Create empty objects for responses only if they did not already exist (avoid overwriting existing responses)
+    responseObject.exist = responseObject.exist || {};
+    responseObject.checkPath = responseObject.checkPath || {};
+    responseObject.rmRF = responseObject.rmRF || {};
+    responseObject.mkdirP = responseObject.mkdirP || {};
+
+    var rmRFSuccessObj = {
+        success: true,
+        message: "foo bar"
+    };
+
+
+    paths.forEach((path) => {
+        responseObject.exist[path] = true;
+        responseObject.checkPath[path] = true;
+        responseObject.rmRF[path] = rmRFSuccessObj;
+        responseObject.mkdirP[path] = true;
+    });
+
+    return responseObject;
+}
```

Comment:
Having to do this setup gets very bulky and obscures the point of tests, so has been moved here.